src/main/java/org/apache/ibatis/reflection/Reflector.java

```diff
@@ -140,35 +144,26 @@ private void addSetMethods(Class<?> clazz) {
   }
 
   private void addMethodConflict(Map<String, List<Method>> conflictingMethods, String name, Method method) {
-    List<Method> list = conflictingMethods.computeIfAbsent(name, k -> new ArrayList<>());
-    list.add(method);
+    if (isValidPropertyName(name)) {
+      List<Method> list = conflictingMethods.computeIfAbsent(name, k -> new ArrayList<>());
+      list.add(method);
+    }
   }
 
   private void resolveSetterConflicts(Map<String, List<Method>> conflictingSetters) {
     for (String propName : conflictingSetters.keySet()) {
       List<Method> setters = conflictingSetters.get(propName);
       Class<?> getterType = getTypes.get(propName);
       Method match = null;
-      ReflectionException exception = null;
       for (Method setter : setters) {
         if (setter.getParameterTypes()[0].equals(getterType)) {
```

Comment:
By this change, when getter methods are conflict, I think can not decide the method that parameter type matched with getterType as best matched method. WDYT?

Reply:
What if we skip the best match check when the getter is ambiguous?
Something like...

```java
boolean isGetterAmbiguous = getMethods.get(propName) instanceof AmbiguousMethodInvoker;
for (Method setter : setters) {
  if (!isGetterAmbiguous && setter.getParameterTypes()[0].equals(getterType)) {
    // should be the best match
```

Reply:
Yes!